x-pack/test/cloud_security_posture_functional/pages/custom_role_access.ts

```diff
@@ -0,0 +1,310 @@
+        benchmark: {
+          id: 'cis_k8s',
+          posture_type: 'kspm',
+          name: 'CIS Kubernetes V1.23',
+          version: 'v1.0.0',
+        },
+        type: 'process',
+      },
+    },
+    {
+      '@timestamp': new Date().toISOString(),
+      resource: { id: cspmResourceId, name: cspmResourceName, sub_type: cspmResourceSubType },
+      result: { evaluation: 'passed' },
+      cloud: {
+        account: {
+          id: '1',
+          name: 'Account 1',
+        },
+      },
+      rule: {
+        name: 'Another upper case rule name',
+        section: 'lower case section',
+        benchmark: {
+          id: 'cis_gcp',
+          posture_type: 'cspm',
+          name: 'CIS Google Cloud Platform Foundation',
+          version: 'v2.0.0',
+        },
+        type: 'process',
+      },
+    },
+    {
+      '@timestamp': new Date().toISOString(),
+      resource: { id: cspmResourceId, name: cspmResourceName, sub_type: cspmResourceSubType },
+      result: { evaluation: 'failed' },
+      cloud: {
+        account: {
+          id: '2',
+          name: 'Account 2',
+        },
+      },
+      rule: {
+        name: 'some lower case rule name',
+        section: 'another lower case section',
+        benchmark: {
+          id: 'cis_gcp',
+          posture_type: 'cspm',
+          name: 'CIS Google Cloud Platform Foundation',
+          version: 'v2.0.0',
+        },
+        type: 'process',
+      },
+    },
+  ];
+
+  const VALID_CUSTOM_ROLE = {
+    name: 'valid_role',
+    elasticsearch: {
+      cluster: [],
+      indices: [
+        {
+          names: ['log-cloud_security_posture.findings-*'],
+          privileges: ['read'],
+        },
+        {
+          names: ['logs-cloud_security_posture.findings_latest-*'],
+          privileges: ['read'],
+        },
+        {
+          names: ['logs-cloud_security_posture.scores-*'],
+          privileges: ['read'],
+        },
+      ],
+      run_as: [],
+    },
+
+    kibana: [
+      {
+        base: ['all'],
+        spaces: ['*'],
+      },
+    ],
+  };
+
+  const INVALID_CUSTOM_ROLE = {
+    name: 'invalid_role',
+    elasticsearch: {
+      cluster: [],
+      indices: [
+        {
+          names: ['log-cloud_security_posture.findings-*'],
+          privileges: ['read'],
+        },
+        {
+          names: ['logs-cloud_security_posture.scores-*'],
+          privileges: ['read'],
+        },
+      ],
+      run_as: [],
+    },
+
+    kibana: [
+      {
+        base: ['all'],
+        spaces: ['*'],
+      },
+    ],
+  };
+
+  const VALID_USER = {
+    name: 'csp_valid_user',
+    password: 'test123',
+  };
+  const INVALID_USER = {
+    name: 'csp_invalid_user',
+    password: 'test123',
+  };
+
+  const navigateToHomePage = async () => {
+    await pageObjects.common.navigateToUrl('app');
+    await pageObjects.header.waitUntilLoadingHasFinished();
+  };
+
+  describe('Custom Role Access', function () {
```

Comment:
nit, IMO these tests are part of each page test suite.

Reply:
done.